pkg/api/customization/authn/user_actions.go

```diff
@@ -136,3 +146,44 @@ func (h *Handler) setPassword(actionName string, action *types.Action, request *
 	request.WriteResponse(http.StatusOK, userData)
 	return nil
 }
+
+func (h *Handler) refreshAttributes(actionName string, action *types.Action, request *types.APIContext) error {
+	isAdmin, err := userIsAdmin(request)
+	if err != nil {
+		return err
+	}
+
+	if !isAdmin {
+		return errors.New("Not Allowed")
+	}
+
+	if request.ID != "" {
+		providerrefresh.TriggerUserRefresh(request.ID, true)
+	} else {
+		providerrefresh.TriggerAllUserRefresh()
+	}
+
+	request.WriteResponse(http.StatusOK, nil)
+	return nil
+}
+
+func userIsAdmin(request *types.APIContext) (bool, error) {
+	store := request.Schema.Store
+	if store == nil {
+		return false, errors.New("no user store available")
+	}
+
+	userID := request.Request.Header.Get("Impersonate-User")
+	if userID == "" {
+		return false, errors.New("can't find user")
+	}
+
+	userData, err := store.ByID(request, request.Schema, userID)
+	if err != nil {
+		return false, err
+	}
+
+	name, ok := userData[client.UserFieldUsername]
+
+	return (ok && (name == "admin")), nil
```

Comment:
Let's grab the global bindings and look for the admin binding. Just looking at the name is waiting for bad things to happen.